packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -189,16 +240,31 @@ export class SelectedNetworkController extends BaseController<
         },
       );
     }
+    // Update the network client ID for the specified domain
     this.#setNetworkClientIdForDomain(domain, networkClientId);
   }
 
-  getNetworkClientIdForDomain(domain: Domain): NetworkClientId {
-    if (this.state.perDomainNetwork) {
-      return this.state.domains[domain] ?? this.state.domains[METAMASK_DOMAIN];
+  getNetworkClientIdForDomain(domain: Domain): NetworkClientId | undefined {
+    if (domain === METAMASK_DOMAIN) {
+      return this.getNetworkClientIdForMetamask();
     }
+    if (!this.state.perDomainNetwork) {
+      return undefined;
```

Comment:
Do we expect anyone to call this method if `perDomainNetwork` is unset? If so, why? Would `undefined` be a safe response?

Reply:
This change was also made under false premises. Currently working on removing them in this branch: https://github.com/MetaMask/core/compare/ad/fix-selected-network-middleware...ad/fix-selected-network-middleware-tweaked?expand=1

Reply:
This change is no longer present in this PR as of https://github.com/MetaMask/core/pull/3908/commits/c78443841150f1f0acf3007fa9d67fae9bce5481